### DISCUSSION TITLE
Proposal for working with volunteer developers in CORE

### DISCUSSION THREAD

**Discussion by drewcdisc** [cdisc-org/cdisc-rules-engine#694] [Norms, process, standards]
At a recent retrospective the CORE team discussed how we want to work with volunteers. This is the process we are proposing.

- All CORE volunteer work begins as a typical [GitHub Issue](https://github.com/features/issues) (that is, not in ZenHub) because volunteers will have access to GitHub but not ZenHub.
- A volunteer will typically [fork the repository](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo) in which they wish to contribute. Before beginning development the volunteer will provide a [draft PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork) that will contain the discussion around their proposed contribution or issue resolution.
- When the CORE team receives a draft PR they will:
	- **Add that PR to Zenhub in the Non-Scrum Team column** (ZenHub pipeline). As ZenHub only creates private ZenHub metadata for the GitHub issue this action will not change what the volunteer sees.
	- **Assign that volunteer to the issue** to clarify who will be working on the issue.
- When the volunteer [asks](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) for a [review of the PR](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews) that request immediately shows up in the Ready to Review column of the CORE Scrum board. At this point the Scrum Team will choose to do one of two things:
	- **Keep the Review request in the Ready to Review column**, thus signaling that they are choosing to complete the Review during the current sprint. NOTE that this also necessitates that the Scrum Team decides whether or not to make room for the review by removing work that is already in the Sprint.
	- **Move the Review request to the Non-Scrum Team column**, thus signaling that the CORE Scrum Team is delaying the completion of the review until the next (or later) Sprint.
- When the Scrum Team decides to take on the volunteer's Review request during the current Sprint, a single Dev Team member will take on the assignment for the Review and work with the volunteer on that issue and shepherd the issue through the process. Although the initial Review assignee has the option of completing the review work alone, additional members of the Scrum Team will participate in the Review as needed .

**Comment by dostiep** [comment]
Just want to clafiry bullet point #2. When a volunteer decides to work on an issue, he/she needs to create a draft PR before any work starts, right? That's what I get from what I read but just want to make sure. I assume creating a draft PR means the issue in question is assigned to the volunteer. Is that enough to make sure no more than one volunteer work on the same issue?

**Reply by drewcdisc** [reply]
To be clear, a volunteer can do what a volunteer will do. If a volunteer wanted to build something in secret without telling anyone they certainly can do that. :)

My understanding is that the GitHub PR is intended to be a conduit where the discussion can happen around a proposed issue resolution. It also makes integration easier later because both CDISC and the volunteer can have more consensus about what is about it be built.

We had not really discussed whether or not multiple volunteers were working on an issue. That would be an exciting problem to have. :) Since you brought it up we probably should ensure that the volunteer is assigned to the issue as part of the process. Thanks for catching that. I'll modify the lead discussion post.
